### PR TITLE
fix javadoc links to method anchors

### PR DESCRIPTION
**Pull request by tyrasd**
argh, this must have changed at some point (~~perhaps with `0.6.0` ?~~ with `0.5.10` actually – when we switched to java11 on the build server??)

We need to replace links ending with `https://docs.ohsome.org/java/oshdb/….html#…-…[-…]*-` to `https://docs.ohsome.org/java/oshdb/….html#…(…[,…]*)`.

## todo

* fix all markdown in documentation folder
  * [x] README.md
  * [x] manual/README.md
  * [x] manual/aggregation.md
  * [x] manual/api.md
  * [x] manual/database-backends.md
  * [x] manual/data-model.md
  * [x] manual/filters.md
  * [x] manual/installation.md
  * [x] manual/map-reduce.md
  * [x] manual/views.md
  * [x] first-steps/README.md
* [x] fix readme
* [x] fix links in other files
* [x] fix links in javadocs (hopefully there are none – they should use javadoc links)
* [x] manually re-publish 0.5.10 docs with the old link schema
* fix links from other projects to the oshdb javadoc (e.g. ohsome-api) [not in this PR, of course]

### CODE REVIEW

File: documentation/manual/filters.md [hunk 1]
```diff
@@ -24,34 +24,34 @@ The OSHDB is able to cope well even with complex polygons that have many vertice
...
+It is possible to [define custom filtering functions](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#osmEntityFilter(org.heigit.bigspatialdata.oshdb.api.generic.function.SerializablePredicate)), that take an OSM entity object as input and decide wether each individual entity should be included in the result or not by returning a boolean value.
 
 _ohsome_ filter
 ---------------
 
-An easy way to provide complex [`filter`s](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#filter-org.heigit.bigspatialdata.oshdb.api.generic.function.String-) is through the functionality of [ohsome filters](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/libs/ohsome-filter#readme), which allow one to define osm data filter in a human readable syntax. With these one can combine several tag, type and geometry filters with arbitrary boolean operators.
```
Comment (tyrasd) [inline documentation/manual/filters.md:50]: note that this also fixes an unrelated bug (`org.heigit.bigspatialdata.oshdb.api.generic.function.String` :arrow_right: `java.lang.String`) in this javadoc link

File: documentation/manual/filters.md [hunk 2]
```diff
@@ -24,34 +24,34 @@ The OSHDB is able to cope well even with complex polygons that have many vertice
...
+An easy way to provide complex [`filter`s](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#filter(java.lang.String)) is through the functionality of [ohsome filters](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/libs/ohsome-filter#readme), which allow one to define osm data filter in a human readable syntax. With these one can combine several tag, type and geometry filters with arbitrary boolean operators.
```
Comment (tyrasd) [inline documentation/manual/filters.md:50]: note that this also fixes an unrelated bug (`org.heigit.bigspatialdata.oshdb.api.generic.function.String` :arrow_right: `java.lang.String`) in this javadoc link

File: documentation/manual/filters.md [hunk 3]
```diff
@@ -24,34 +24,34 @@ The OSHDB is able to cope well even with complex polygons that have many vertice
...
+Filters OSM entities by their OSM data type. These types are: node, way and relation, represented in the OSHDB by the enumeration [OSMType](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/osm/OSMType.html). In a MapReducer, the [method](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#osmType(java.util.EnumSet)) [`osmType`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#osmType(org.heigit.bigspatialdata.oshdb.osm.OSMType,org.heigit.bigspatialdata.oshdb.osm.OSMType...)) allows to filter objects that are of a specific OSM data type.
```
Comment (joker234) [inline documentation/manual/filters.md:34]:
```suggestion
Filters OSM entities by their OSM data type. These types are: node, way and relation, represented in the OSHDB by the enumeration [OSMType](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/osm/OSMType.html). In a MapReducer, the [method](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#osmTypes(java.util.EnumSet)) [`osmType`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#osmType(org.heigit.bigspatialdata.oshdb.osm.OSMType,org.heigit.bigspatialdata.oshdb.osm.OSMType...)) allows to filter objects that are of a specific OSM data type.
```

File: documentation/manual/map-reduce.md [hunk 4]
```diff
@@ -10,14 +10,14 @@ For many of the most frequently used reduce operations, such as the summing up o
...
 
 > For example, when calculating the length (which is a floating point number) of an entity snapshot, the underlying MapReducer changes from type `MapReducer<OSMEntitySnapshot>` to being a `MapReducer<Double>`.
 
 flatMap
 -------
 
-A [`flatMap`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#flatMap-org.heigit.bigspatialdata.oshdb.api.generic.function.SerializableFunction-) operation allows one to map any input value to an arbitrary amount of output values. Each of the output values can be transformed in further map steps individually.
+A [`flatMap`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#flatMap(org.heigit.bigspatialdata.oshdb.api.generic.function.SerializableFunction) operation allows one to map any input value to an arbitrary amount of output values. Each of the output values can be transformed in further map steps individually.
```
Comment (joker234) [inline documentation/manual/map-reduce.md:20]:
```suggestion
A [`flatMap`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html#flatMap(org.heigit.bigspatialdata.oshdb.api.generic.function.SerializableFunction)) operation allows one to map any input value to an arbitrary amount of output values. Each of the output values can be transformed in further map steps individually.
```

File: documentation/manual/views.md [hunk 5]
```diff
@@ -13,7 +13,7 @@ The snapshot view is particularly useful for analysing how the amount of OSM dat
...
+Both views can be used in the OSHDB API in very similar ways and only differ in the type of data that is returned by the [`MapReducer`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html) object that is returned when calling the [`on`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/OSMContributionView.html#on(org.heigit.bigspatialdata.oshdb.api.db.OSHDBDatabase) method of the respective view: The `OSMEntitySnapshotView` returns a MapReducer of [`OSMEntitySnapshot`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/object/OSMEntitySnapshot.html) objects, while the `OSMContributionView` returns a MapReducer of [`OSMContribution`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/object/OSMContribution.html) objects.
```
Comment (joker234) [inline documentation/manual/views.md:16]:
```suggestion
Both views can be used in the OSHDB API in very similar ways and only differ in the type of data that is returned by the [`MapReducer`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.html) object that is returned when calling the [`on`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/mapreducer/OSMContributionView.html#on(org.heigit.bigspatialdata.oshdb.api.db.OSHDBDatabase)) method of the respective view: The `OSMEntitySnapshotView` returns a MapReducer of [`OSMEntitySnapshot`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/object/OSMEntitySnapshot.html) objects, while the `OSMContributionView` returns a MapReducer of [`OSMContribution`](https://docs.ohsome.org/java/oshdb/0.6.0/aggregated/org/heigit/bigspatialdata/oshdb/api/object/OSMContribution.html) objects.
```

**Review by joker234 — CHANGES_REQUESTED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**